heron/spi/src/java/com/twitter/heron/spi/statemgr/IStateManager.java

```diff
@@ -276,4 +277,27 @@ public String getName() {
    */
   ListenableFuture<PhysicalPlans.PhysicalPlan> getPhysicalPlan(
       WatchCallback watcher, String topologyName);
+
+  /**
+   * Set the Stateful Checkpoint
+   *
+   * @return Boolean - Success or Failure
+   */
+  ListenableFuture<Boolean> setStatefulCheckpoint(
+      CheckpointManager.StatefulConsistentCheckpoints checkpoint, String topologyName);
+
+  /**
+   * Get the Stateful Checkpoint
+   *
+   * @return Boolean - Success or Failure
+   */
+  ListenableFuture<CheckpointManager.StatefulConsistentCheckpoints> getStatefulCheckpoint(
```

Comment:
We should store the checkpoints in a way that allows us to easily delete (or manually browse when troubleshooting) an entire set by the packing plan hash and checkpoint id, so we should considering including those in the path in the directory tree.

Consider the use case where the topology gets redeployed with the same parallelism and packing plan hash. We can reuse that state with the new topology. If instead the topology is deployed with different parallelism (i.e., a new packing plan), we can not reload that old state directly. We also might want to easily prune all old checkpoints bound to old packing plans. Hence, being able to delete an entire file system branch would be ideal.